backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -790,6 +791,32 @@ void ExternConverter_action_selector::convertExternInstance(
     ctxt->action_profiles->append(action_profile);
 }
 
+Util::IJson* ExternConverter_log_msg::convertExternFunction(
+    UNUSED ConversionContext* ctxt, UNUSED const P4::ExternFunction* ef,
+    UNUSED const IR::MethodCallExpression* mc, UNUSED const IR::StatOrDecl* s,
```

Comment:
would be nice if you could somehow factor the common code between the parser and this in a single function.